### PR TITLE
Add aten::hardtanh/hardswish and their variants

### PR DESCRIPTION
**Pull request by fengyuan14**

### CODE REVIEW

File: test/xpu/fin_grain/test_ops_xpu.py [hunk 1]
```diff
@@ -21,7 +21,9 @@
     slowTest,
     unMarkDynamoStrictTest,
 )
-
+import os
+base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+sys.path.append(base_dir)
```
Comment (fengyuan14) [inline test/xpu/fin_grain/test_ops_xpu.py:26]: Why this?

File: src/aten/sycl/ActivationHardswishKernel.h
Comment (fengyuan14) [inline src/aten/sycl/ActivationHardswishKernel.h:1]: `ActivationHardswishKernels.h`

File: src/aten/sycl/ActivationHardtanhKernel.h
Comment (fengyuan14) [inline src/aten/sycl/ActivationHardtanhKernel.h:1]: `ActivationHardtanhKernels.h`

File: src/aten/Activation.cpp [hunk 2]
```diff
@@ -158,4 +160,90 @@ Tensor& XPUNativeFunctions::gelu_backward_out(
...
+  return at::hardtanh_out(self, self, min, max);
+}
+
+Tensor& XPUNativeFunctions::hardtanh_backward_out(
+    const Tensor& grad_output,
+    const Tensor& self,
+    const Scalar& min,
+    const Scalar& max,
+    Tensor& grad_input) {
+  auto iter = TensorIterator::borrowing_binary_op(grad_input, grad_output, self);
```
Comment (fengyuan14) [inline src/aten/Activation.cpp:207]: Please also review CUDA generated code to rebase all operators level codes. `build/aten/src/ATen/RegisterCUDA.cpp`. The file is generated at building time. You can build a PyTorch with CUDA or review a built project.
image

File: test/xpu/fin_grain/test_ops_xpu.py (hunk 1 above)
Comment (yucai-intel) [reply test/xpu/fin_grain/test_ops_xpu.py]: This is for running the code locally and has been deleted.

**Comment by xytintel** [comment]
Please follow the standard file naming conventions.

**Review by xytintel — APPROVED** [review]

**Pull request merged**